core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -380,14 +397,24 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
     sycl::queue& q = *instance.m_queue;
 
     q.submit([&](sycl::handler& cgh) {
-      const auto shmem_begin  = m_shmem_begin;
-      const auto shmem_size   = m_shmem_size;
-      const auto scratch_size = m_scratch_size[1];
-      auto team_lambda        = [=](sycl::nd_item<1> item) {
-        // FIXME_SYCL Add scratch memory
-        const member_type team_member(nullptr, shmem_begin, shmem_size, nullptr,
-                                      scratch_size, item);
-
+      const auto shmem_begin     = m_shmem_begin;
+      const int scratch_size[2]  = {m_scratch_size[0], m_scratch_size[1]};
+      void* const scratch_ptr[2] = {m_scratch_ptr[0], m_scratch_ptr[1]};
+
+      // FIXME_SYCL accessors seem to need a size greater than zero at least for
+      // host queues
+      sycl::accessor<char, 1, sycl::access::mode::read_write,
+                     sycl::access::target::local>
+          team_scratch_memory_L0(
+              sycl::range<1>(std::max(m_scratch_size[0] + shmem_begin, 1)),
+              cgh);
```

Comment:
Any particular reason to have this here rather than in the constructor and assigning the right address to `m_scratch_ptr`?

Reply:
Yes, the `sycl::handler` `cgh` is only available in this context.